src/sidebar/components/menu.js

```diff
@@ -96,7 +106,11 @@ export default function Menu({
   // It should also close if the user presses a key which activates menu items.
   const handleMenuKeyDown = event => {
     if (event.key === 'Enter' || event.key === ' ') {
-      closeMenu();
+      // Don't close the menu right away, the link still needs to be present at least one
+      // more render cycle for them to be followed.
```

Comment:
Well, more specifically I think this works because the event bubbles up when you call setTimeout allowing it be handled before the anchor is deleted. I demonstrated this here 
https://jsfiddle.net/ft8z1wux/7
How or why that happens in a browser is not clear to me, so we should change that comment around